File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    score.sort(reverse=True)
    
    temp = []
    answer = 0
    for i in score:
        if len(temp) == m:
            answer += min(temp) * m
            temp = []
        
        temp.append(i)
        
    if len(temp) == m:
        answer += min(temp) * m
        
    return answer

File: 프로그래머스/1/135808. 과일 장수/과일 장수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int compare(const void* a, const void* b) {
    return *(int*)b - *(int*)a;
}

int solution(int k, int m, int score[], size_t score_len) {
    qsort(score, score_len, sizeof(int), compare);
    
    int answer = 0;
    for (int i = m - 1; i < score_len; i += m) {
        answer += score[i] * m;
    }
    
    return answer;
}